File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        Map<Integer,Integer> prefix = new HashMap<>();
        int sum = 0;
        prefix.put(0,0);
        for(int i = 0 ; i < nums.length; i++) {
            sum += nums[i];
            if(!prefix.containsKey(sum%k))
                prefix.put(sum%k,i+1);
            else if(prefix.get(sum%k) < i)
                return true;
            
        }
        
        return false;
    }
}

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var checkSubarraySum = function(nums, k) {
    if(nums.length === 1) return false;
    let sum = 0;
    const map = new Map();
    map.set(0,0);

    for(let i =0; i < nums.length; i++) {
        sum += nums[i];
        if(map.has(sum % k)) {
            if(map.get(sum % k) < i)
            return true;
        } else
        map.set(sum % k,i+1)
    }

    return false;
};